### PR TITLE
EVG-2803 Fix generate tasks bugs

### PR DESCRIPTION
**Pull request by bsamek**

### CODE REVIEW

File: util/json.go [hunk 1]
```diff
@@ -13,9 +13,9 @@ func ReadJSONInto(r io.ReadCloser, data interface{}) error {
 	defer r.Close()
 	bytes, err := ioutil.ReadAll(r)
 	if err != nil {
-		return errors.Wrap(err, "error reading JSON")
+		return errors.Wrapf(err, "error reading JSON (%s)", string(bytes))
 	}
-	return errors.Wrapf(json.Unmarshal(bytes, data), "error attempting to unmarshal into %T: %s", data, string(bytes[:]))
+	return errors.Wrapf(json.Unmarshal(bytes, data), "error attempting to unmarshal into %T: %s", data, string(bytes))
```
Comment (tychoish) [inline util/json.go]: do we worry about leaking data in an awkward way?

File: model/project_parser.go [hunk 2]
```diff
@@ -188,12 +188,21 @@ func (vs *variantSelector) UnmarshalYAML(unmarshal func(interface{}) error) erro
 		return err
 	}
 	if len(md) == 0 {
-		return errors.New("variant selector must not be empty")
+		md = nil
```
[the remark this answers is not in the document]
Comment (bsamek) [reply model/project_parser.go]: Given my other changes, this is unnecessary. Reverted.

File: command/generate.go [hunk 3]
```diff
@@ -52,7 +53,7 @@ func (c *generateTask) Execute(ctx context.Context, comm client.Communicator, lo
 	if catcher.HasErrors() {
 		return errors.WithStack(catcher.Resolve())
 	}
-	post := makeJsonOfAllFiles(jsonBytes)
+	post := makeJsonOfAllFiles(jsonBytes, logger)
```
[the remark this answers is not in the document]
Comment (bsamek) [reply command/generate.go]: done

File: util/json.go (hunk 1 above)
Comment (bsamek) [reply util/json.go]: fair, removed

File: rest/data/generate_test.go [hunk 4]
```diff
@@ -167,4 +175,12 @@ func TestGenerateTasks(t *testing.T) {
...
 		assert.True(v)
 	}
+	v, err := version.FindOneId("sample_version")
+	assert.NoError(err)
+	p := model.Project{}
+	err = model.LoadProjectInto([]byte(v.Config), "mci", &p)
+	assert.NoError(err)
+	assert.Len(p.Tasks, 4)
+	assert.Len(p.BuildVariants[0].Tasks, 1)
+	assert.Len(p.BuildVariants[1].Tasks, 4)
```
Comment (tychoish) [inline rest/data/generate_test.go]: nit: also check that you didn't update sample base project.

**Pull request merged**